codersdk/groups.go

```diff
@@ -35,6 +35,18 @@ type Group struct {
 	Source         GroupSource   `json:"source"`
 }
 
+type ReducedGroup struct {
+	ID             uuid.UUID `json:"id" format:"uuid"`
+	Name           string    `json:"name"`
+	DisplayName    string    `json:"display_name"`
+	OrganizationID uuid.UUID `json:"organization_id" format:"uuid"`
+	// How many users are in this group. Shows the total count,
+	// even if the user is not authorized to read group member details.
+	// May be greater than `len(Group.Members)`.
+	TotalMemberCount int    `json:"total_member_count"`
+	AvatarURL        string `json:"avatar_url"`
+}
+
```

Comment:
Leaving a larger comment about potentially making this a database model based on a better `group_member` view.